server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskAction.java

```diff
@@ -103,4 +118,40 @@ private void checkPermission(@Nullable String projectUuid) {
       throw insufficientPrivilegesException();
     }
   }
+
+  private static void maskErrorStacktrace(Request wsRequest, CeActivityDto ceActivityDto) {
+    Set<AdditionalField> fromRequest = TaskAction.AdditionalField.getFromRequest(wsRequest);
+    if (!fromRequest.contains(TaskAction.AdditionalField.STACKTRACE)) {
+      ceActivityDto.setErrorStacktrace(null);
+    }
+  }
+
+  private enum AdditionalField {
+    STACKTRACE;
+
+    public static Set<AdditionalField> getFromRequest(Request wsRequest) {
+      List<String> strings = wsRequest.paramAsStrings(PARAM_ADDITIONAL_FIELDS);
+      if (strings == null) {
+        return Collections.emptySet();
+      }
+      return strings.stream()
+        .map(s -> {
+          for (AdditionalField field : AdditionalField.values()) {
+            if (field.name().equalsIgnoreCase(s)) {
+              return field;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 17. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)